### DISCUSSION TITLE
Has anyone tried react-navigation ServerContainer rendering as SPA in next?

### DISCUSSION THREAD

**Discussion by tonyxiao** [nandorojo/solito#105] [General]
New to this, got a Expo web app and want to do static / server side rendering and experimenting.

has anyone seen the following approach? It seems to work ok in a mini-test app but I'm not certain if anyone else tried it before. 

<img width="738" alt="image" src="https://user-images.githubusercontent.com/696842/178090181-7000b5fd-7560-4e32-89a8-1358b2c639e8.png">

**Comment by nandorojo** [comment]
that approach will actually break react hydration. the server and client need to render the same thing at the first render

**Comment by tonyxiao** [comment]
Hmm isn't server-container designed to solve this problem, with or without next.js? Where would the hydration delta come from?

**Comment by nandorojo** [comment]
For what it's worth, with Solito, you don't use react navigation on Web, so this approach wouldn't apply to the Solito architecture.
